sklearn/preprocessing/label.py

```diff
@@ -889,16 +890,24 @@ def transform(self, y):
             A matrix such that `y_indicator[i, j] = 1` iff `classes_[j]` is in
             `y[i]`, and 0 otherwise.
         """
+        self._cached_dict = None
         check_is_fitted(self, 'classes_')
 
-        class_to_index = dict(zip(self.classes_, range(len(self.classes_))))
+        class_to_index = self._cached_dict_for()
         yt = self._transform(y, class_to_index)
 
         if not self.sparse_output:
             yt = yt.toarray()
 
         return yt
 
+    def _cached_dict_for(self):
+        if (self._cached_dict is None):
+            self._cached_dict = dict(zip(self.classes_, \
```

Comment:
The `\` is not necessary, since you have an open parenthesis.
You also need to align the arguments to comply with PEP8:
```py
            self._cached_dict = dict(zip(self.classes_,
                                         range(len(self.classes_))))
```

Reply:
removed